File: src/src/bmkim/level_2/타겟넘버.java
public class 타겟넘버 {
    static int answer = 0;
    public static int solution(int[] numbers, int target) {
        DFS(numbers, target, 0, 0);
        return answer;
    }

    public static void DFS(int[] numbers, int target, int index, int sum) {
        if (index == numbers.length) {
            if (target == sum) {
                answer++;
            }
        } else {
            DFS(numbers, target, index+1, sum+numbers[index]);
            DFS(numbers, target, index+1, sum-numbers[index]);
        }
    }
}

File: src/src/yjhwang/level_2/타겟넘버.js
function solution(numbers, target) {
  let count = 0;

  function dfs(sum, index) {
    if (numbers.length === index) {
      if (sum === target) {
        count += 1;
      }
      return;
    }

    dfs(sum + numbers[index], index + 1);
    dfs(sum - numbers[index], index + 1);
  }

  dfs(0, 0);

  return count;
}

// console.log(solution([1, 1, 1, 1, 1], 3));
console.log(solution([4, 1, 2, 1], 4));